Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -716,21 +727,14 @@ void AODProducerWorkflowDPL::fillMCParticlesTable(o2::steer::MCKinematicsReader&
           mToStore[Triplet_t(source, event, particle)] = 1;
           // treating contributors of global tracks
           auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
-          if (contributorsGID[GIndex::Source::ITS].isIndexSet() && contributorsGID[GIndex::Source::TPC].isIndexSet()) {
-            auto mcTruthITS = data.getTrackMCLabel(contributorsGID[GIndex::Source::ITS]);
-            if (mcTruthITS.isValid()) {
-              source = mcTruthITS.getSourceID();
-              event = mcTruthITS.getEventID();
-              particle = mcTruthITS.getTrackID();
-              mToStore[Triplet_t(source, event, particle)] = 1;
-            }
-            auto mcTruthTPC = data.getTrackMCLabel(contributorsGID[GIndex::Source::TPC]);
-            if (mcTruthTPC.isValid()) {
-              source = mcTruthTPC.getSourceID();
-              event = mcTruthTPC.getEventID();
-              particle = mcTruthTPC.getTrackID();
-              mToStore[Triplet_t(source, event, particle)] = 1;
-            }
+          if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+            markMCTrackForSrc(contributorsGID, GIndex::Source::TPC);
+          }
+          if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
+            markMCTrackForSrc(contributorsGID, GIndex::Source::ITS);
+          }
+          if (contributorsGID[GIndex::Source::TOF].isIndexSet()) {
+            markMCTrackForSrc(contributorsGID, GIndex::Source::TOF);
```

Comment:
@nburmaso sorry, I got some doubts about this call: the `contributorsGID[GIndex::Source::TOF]` refers to TOF cluster, then in the `markMCTrackForSrc` you are using `data.getTrackMCLabel` with this reference. But the TOF cluster's MC labels are not coming from the span with 1 label per cluster as  https://github.com/AliceO2Group/AliceO2/blob/9e0753e352b7b8279d80a1efae3af30493d20edd/DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h#L438 expects but are stored in https://github.com/AliceO2Group/AliceO2/blob/9e0753e352b7b8279d80a1efae3af30493d20edd/DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h#L305 with possibly multiple labels per cluster. That's why the loop https://github.com/AliceO2Group/AliceO2/blob/f757cc5defab87b181d9afab15ad7f8fd64acf7d/Detectors/GlobalTracking/src/MatchTOF.cxx#L1232-L1236 is used to decide if the match is fake or not.
I think the right thing would be to have a fix by Francesco and then to use isFake result of the TOF match.
Otherwise, one has to use https://github.com/AliceO2Group/AliceO2/blob/9e0753e352b7b8279d80a1efae3af30493d20edd/DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainer.h#L607 to access TOF clusters and implement the loop over cluster contributors.